From 099650168b4512760d9ce0a438a563672544736c Mon Sep 17 00:00:00 2001
From: b02550 <b02550@freescale.com>
Date: Fri, 22 May 2009 09:16:47 +0800
Subject: [PATCH 059/280] ENGR00112570 IPUv3: Avoid ipu limitation by adding
 check

Add ipu limitation checking for the input&overlay parameters.

Signed-off-by: DingQiang <b02550@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index ac7e87f..748f1ba 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -399,6 +399,7 @@ static int _ipu_task_check(ipu_lib_input_param_t * input,
 		if ((input->input_crop_win.win_h + input->input_crop_win.pos.y) > input->height)
 			input->input_crop_win.win_h = input->height - input->input_crop_win.pos.y;
 		ipu_priv_handle->iwidth = input->input_crop_win.win_w;
+		ipu_priv_handle->iwidth = ipu_priv_handle->iwidth  - ipu_priv_handle->iwidth%8;
 		ipu_priv_handle->iheight = input->input_crop_win.win_h;
 
 		if ((ipu_priv_handle->iwidth != input->width) || (ipu_priv_handle->iheight != input->height)) {
@@ -448,6 +449,7 @@ static int _ipu_task_check(ipu_lib_input_param_t * input,
 		}
 	} else {
 		ipu_priv_handle->iwidth = input->width;
+		ipu_priv_handle->iwidth = ipu_priv_handle->iwidth - ipu_priv_handle->iwidth%8;
 		ipu_priv_handle->iheight = input->height;
 	}
 
@@ -458,6 +460,7 @@ static int _ipu_task_check(ipu_lib_input_param_t * input,
 			if ((overlay->ov_crop_win.win_h + overlay->ov_crop_win.pos.y) > overlay->height)
 				overlay->ov_crop_win.win_h = overlay->height - overlay->ov_crop_win.pos.y;
 			ipu_priv_handle->ovwidth = overlay->ov_crop_win.win_w;
+			ipu_priv_handle->ovwidth = ipu_priv_handle->ovwidth  - ipu_priv_handle->ovwidth%8;
 			ipu_priv_handle->ovheight = overlay->ov_crop_win.win_h;
 
 			if ((ipu_priv_handle->ovwidth != overlay->width) || (ipu_priv_handle->ovheight != overlay->height)) {
@@ -507,6 +510,7 @@ static int _ipu_task_check(ipu_lib_input_param_t * input,
 			}
 		} else {
 			ipu_priv_handle->ovwidth = overlay->width;
+			ipu_priv_handle->ovwidth = ipu_priv_handle->ovwidth - ipu_priv_handle->ovwidth%8;
 			ipu_priv_handle->ovheight = overlay->height;
 		}
 	}
-- 
1.8.0

